pebb.so/src/action.cpp

```diff
@@ -73,51 +73,71 @@ pebb_action::~pebb_action() {
  * the module's properties collection
  * @return true if no fatal error occured, false otherwise
  */
-bool pebb_action::get_all_pebb_config_keys(void) {;
+bool pebb_action::get_all_pebb_config_keys(void) {
   string msg;
   int error;
   bool bsts = true;
 
   RVSTRACE_
 
   if (property_get("host_to_device", &prop_h2d, true)) {
-      msg = "invalid 'host_to_device' key";
-      rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
-      bsts = false;
+    msg = "invalid 'host_to_device' key";
+    rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
+    bsts = false;
   }
 
   if (property_get("device_to_host", &prop_d2h, true)) {
-      msg = "invalid 'device_to_host' key";
-      rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
-      bsts = false;
+    msg = "invalid 'device_to_host' key";
+    rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
+    bsts = false;
   }
 
   error = property_get_uint_list<uint32_t>(RVS_CONF_BLOCK_SIZE_KEY,
-                                   YAML_DEVICE_PROP_DELIMITER,
-                                   &block_size, &b_block_size_all);
+      YAML_DEVICE_PROP_DELIMITER,
+      &block_size, &b_block_size_all);
   if (error == 1) {
-      msg = "invalid '" + std::string(RVS_CONF_BLOCK_SIZE_KEY) + "' key";
-      rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
-      bsts = false;
+    msg = "invalid '" + std::string(RVS_CONF_BLOCK_SIZE_KEY) + "' key";
+    rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
+    bsts = false;
   } else if (error == 2) {
     b_block_size_all = true;
     block_size.clear();
   }
 
-  error = property_get_int<uint32_t>
-  (RVS_CONF_B2B_BLOCK_SIZE_KEY, &b2b_block_size);
+  error = property_get_int<uint32_t>(RVS_CONF_B2B_BLOCK_SIZE_KEY, &b2b_block_size);
   if (error == 1) {
     msg = "invalid '" + std::string(RVS_CONF_B2B_BLOCK_SIZE_KEY) + "' key";
     rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
-      bsts = false;
+    bsts = false;
   }
 
   error = property_get_int<int>(RVS_CONF_LINK_TYPE_KEY, &link_type);
   if (error == 1) {
     msg = "invalid '" + std::string(RVS_CONF_LINK_TYPE_KEY) + "' key";
     rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
-      bsts = false;
+    bsts = false;
+  }
+
+  error = property_get_int<uint32_t>(RVS_CONF_HOT_CALLS_KEY, &hot_calls, DEFAULT_HOT_CALLS);
+  if (error == 1) {
+    msg = "invalid '" + std::string(RVS_CONF_HOT_CALLS_KEY) + "' key";
+    rvs::lp::Err(msg, MODULE_NAME_CAPS, action_name);
+    bsts = false;
+  }
+
+  error = property_get_int<uint32_t>(RVS_CONF_WARM_CALLS_KEY, &warm_calls, DEFAULT_WARM_CALLS);
```

Comment:
should there be a check on conditions for hot and warm calls? if one should be greater than the other etc?

Reply:
warm and hot calls shall not be 0 - always default value.